pyogrio/geopandas.py

```diff
@@ -5,7 +5,7 @@
 
 
 def read_dataframe(
-    path,
+    path_or_buffer,
```

Comment:
Do we mind that this a breaking change? If you currently specify `read_dataframe(path="my/path"), this will break it.

Reply:
I can also change it back (had the same hesitation when changing it). Or, while we are at it, we could also make this first argument a positional-only argument. At least when we change it to `path_or_buffer`, I would do that. If it's just path, it might be nice one can do `path=..`

Reply:
It seems very reasonable for this to be a position-only argument.

Reply:
When also allowing file-like objects, the `path_or_buffer` is maybe also not fully descriptive .. (we could also take something more generic like "source")

Reply:
My concern with `source` is that we want to keep the concept of an OGR Datasource as a term; here it is either the path / URI to or bytes that would be used as input to that source...  `path_or_buffer` seems like a reasonable compromise.

Reply:
Yes, it's just that a file-like object (that has a `read()` method) is strictly speaking *not yet* a bytes buffer

Reply:
Right.  Names are hard.  `path_or_readable`?  Or just `path`, acknowledging that it isn't always strictly a filesystem path.

Reply:
Yeah, just keeping it as `path` is indeed maybe also fine. I could still make it a positional-only keyword, then the name doesn't matter that much

Reply:
> positional-only keyword

still seems fine to me; should always be the first parameter rather than from keyword.